lightninglabs/taproot-assets
# monitoring: add prometheus metrics collector

Add monitoring package. Currently, the only registered metrics are for the grpc server interface.

We did not follow the same route as lnd, where there is a external process collecting and exposing most of the metrics (`lndmon`). It will be tapd itself the one that will register and expose the metrics.

How to test this PR?
- Make sure the binary is build with the `monitoring` tag (`dev` includes it by default)
- Make sure `prometheus.active` is set to true in the config/cli tapd params
- Hit the prometheus server port (`8989` by default)


Fixes #298
